crates/core/src/db/queries/metadatas.rs

```diff
@@ -435,6 +438,193 @@ pub fn list(
         .context("Failed to load nft(s)")
 }
 
+/// Input parameters for the [`wallet_nfts`] query.
+#[derive(Debug)]
+pub struct WalletNftOptions {
+    /// wallet address
+    pub wallet: String,
+    /// Auction house of the collection
+    pub auction_house: Option<String>,
+    /// Marketplace program in which the collection is listed
+    pub marketplace_program: Option<String>,
+    /// nft in one or more specific collections
+    pub collections: Option<Vec<String>>,
+    /// nfts listed for sale
+    pub listed: Option<bool>,
+    /// Sort by Price or Listed at
+    pub sort_by: Option<Sort>,
+    /// Order the resulting rows by 'Asc' or 'Desc'
+    pub order: Option<Order>,
+    /// Limit the number of returned rows
+    pub limit: u64,
+    /// Skip the first `n` resulting rows
+    pub offset: u64,
+}
+
+impl From<Sort> for Listings {
+    fn from(sort: Sort) -> Self {
+        match sort {
+            Sort::Price => Listings::Price,
+            Sort::ListedAt => Listings::CreatedAt,
+        }
+    }
+}
+
+/// Handles queries for a wallet Nfts
+///
+/// # Errors
+/// returns an error when the underlying queries throw an error
+#[allow(clippy::too_many_lines)]
+pub fn wallet_nfts(conn: &Connection, options: WalletNftOptions) -> Result<Vec<Nft>> {
+    let WalletNftOptions {
+        wallet,
+        auction_house,
+        marketplace_program,
+        collections,
+        listed,
+        sort_by,
+        order,
+        limit,
+        offset,
+    } = options;
+
+    let sort_unwrap = sort_by.map_or(Listings::Price, Into::into);
+
+    let order_unwrap = order.unwrap_or(Order::Desc);
+
+    let current_time = Utc::now().naive_utc();
+
+    let mut listings_query = Query::select()
+        .columns(vec![
+            (Listings::Table, Listings::Metadata),
+            (Listings::Table, Listings::Price),
+            (Listings::Table, Listings::Seller),
+        ])
+        .from(Listings::Table)
+        .order_by((Listings::Table, sort_unwrap), order_unwrap)
+        .cond_where(
+            Condition::all()
+                .add(Expr::tbl(Listings::Table, Listings::PurchaseId).is_null())
+                .add(Expr::tbl(Listings::Table, Listings::CanceledAt).is_null())
+                .add(
+                    Expr::tbl(Listings::Table, Listings::AuctionHouse)
+                        .ne(pubkeys::OPENSEA_AUCTION_HOUSE.to_string()),
+                )
+                .add(
+                    Expr::tbl(Listings::Table, Listings::Expiry)
+                        .is_null()
+                        .or(Expr::tbl(Listings::Table, Listings::Expiry).gt(current_time)),
+                ),
+        )
+        .take();
```

Comment:
shouldn't need the lateral_join again use left_join. Just move all the conditions to the main query.